### PR TITLE
PWN-3081 - Fixed disabled broken state on HomeScreen

### PR DESCRIPTION
**Pull request by kabramovp2p**
## Jira Ticket

https://p2pvalidator.atlassian.net/browse/PWN-3081

## Description of Work

fix disabled broken state by switching single -> factory

### CODE REVIEW

File: app/src/main/java/org/p2p/wallet/home/HomeModule.kt [hunk 1]
```diff
@@ -53,7 +53,16 @@ object HomeModule : InjectionModule {
         factory { HomeDatabaseRepository(get()) } bind HomeLocalRepository::class
 
         /* Cached data exists, therefore creating singleton */
-        single { HomePresenter(get(), get(), get(), get(), get(), get()) } bind HomeContract.Presenter::class
+        factory {
+            HomePresenter(
+                updatesManager = get(),
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/home/HomeModule.kt:58]: wow, nice!

File: app/src/main/java/org/p2p/wallet/home/ui/main/adapter/TokenAdapter.kt [hunk 2]
```diff
@@ -17,7 +17,7 @@ class TokenAdapter(
 
     fun setItems(new: List<HomeElementItem>, isZerosHidden: Boolean, state: VisibilityState) {
         this.isZerosHidden = isZerosHidden
-        val old = ArrayList(data)
+        val old = data.toMutableList()
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/home/ui/main/adapter/TokenAdapter.kt:20]: `ArrayList(data)` - is a way of deep copying all the elements and creating a new reference. 
I'm not sure that `.toMutableList()` can achieve the same.

In my opinion, when you call `data.clear()` - `old` will clear too.

File: app/src/main/java/org/p2p/wallet/intercom/IntercomService.kt [hunk 3]
```diff
@@ -12,7 +13,7 @@ object IntercomService {
         Intercom.client().setInAppMessageVisibility(Intercom.Visibility.GONE)
     }
 
-    fun signIn(userId: String, onMessageReceived: (count: Int) -> Unit) {
+    fun signIn(userId: String, onMessageReceived: (count: Int) -> Unit = { NoOp }) {
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/intercom/IntercomService.kt:16]: parameter is not used

**Review by gslevinkov — CHANGES_REQUESTED** [review]

File: app/src/main/java/org/p2p/wallet/home/ui/main/adapter/TokenAdapter.kt (hunk 2 above)
Comment (kabramovp2p) [reply app/src/main/java/org/p2p/wallet/home/ui/main/adapter/TokenAdapter.kt:20]: ???
image

Comment (kabramovp2p) [reply app/src/main/java/org/p2p/wallet/home/ui/main/adapter/TokenAdapter.kt:20]: i presume it is same as ArrayList(data)  ? isn't it ?

Comment (gslevinkov) [reply app/src/main/java/org/p2p/wallet/home/ui/main/adapter/TokenAdapter.kt:20]: damn, yes it is. sorry

Comment (gslevinkov) [reply app/src/main/java/org/p2p/wallet/home/ui/main/adapter/TokenAdapter.kt:20]: damn, yes it is. sorry

**Review by 1sakov — APPROVED** [review]

**Review by gslevinkov — APPROVED** [review]

**Review by davran1001 — APPROVED** [review]

**Pull request merged**